server/container/guice/memory-guice/src/test/java/org/apache/james/MemoryJmapJamesServerTest.java

```diff
@@ -19,20 +19,85 @@
 
 package org.apache.james;
 
+import static org.apache.james.JmapJamesServerContract.DOMAIN_LIST_CONFIGURATION_MODULE;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.apache.james.jmap.JMAPConfiguration;
 import org.apache.james.mailbox.extractor.TextExtractor;
 import org.apache.james.mailbox.store.search.PDFTextExtractor;
 import org.apache.james.modules.TestJMAPServerModule;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-class MemoryJmapJamesServerTest implements JmapJamesServerContract {
+class MemoryJmapJamesServerTest {
+
     private static final int LIMIT_TO_10_MESSAGES = 10;
 
-    @RegisterExtension
-    static JamesServerExtension jamesServerExtension = new JamesServerBuilder()
-        .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
-            .combineWith(MemoryJamesServerMain.IN_MEMORY_SERVER_AGGREGATE_MODULE)
-            .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
-            .overrideWith(binder -> binder.bind(TextExtractor.class).to(PDFTextExtractor.class))
-            .overrideWith(DOMAIN_LIST_CONFIGURATION_MODULE))
-        .build();
+    @Nested
+    class JmapJamesServerTest implements JmapJamesServerContract {
+
+        @RegisterExtension
+        JamesServerExtension jamesServerExtension = new JamesServerBuilder()
```

Comment:
you can put the builder into a field and just call `build` for this nested class.
That way, BadAliasKeyStore can reuse the builder and just add the keystore override.